ruint-macro/src/lib.rs

```diff
@@ -120,19 +147,19 @@ fn parse(value: &str, bits: &str) -> Result<(usize, Vec<u64>), String> {
     // Check value range
     if limbs.len() > num_limbs || limbs.last().copied().unwrap_or(0) > mask {
         let value = value.trim_end_matches('_');
-        return Err(format!("Value too large for Uint<{bits}>: {value}"));
+        return Err(format!("Value too large for {base_type}<{bits}>: {value}"));
     }
 
     Ok((bits, limbs))
 }
 
 /// Transforms a [`Literal`] and returns the substitute [`TokenTree`]
-fn transform_literal(literal: Literal) -> TokenTree {
+fn transform_literal(literal: Literal, base_type: LiteralBaseType) -> TokenTree {
     let source = literal.to_string();
-    if let Some((value, bits)) = source.split_once('U') {
-        let tokens = parse(value, bits).map_or_else(
+    if let Some((value, bits)) = source.split_once(base_type.delimiter()) {
```

Comment:
When `base_type` is `B` this will split on the first `B`, which may be a valid hex digit! We should use `rsplit_once` instead to start searching from the back. But even then there is potential for confusion:
`0x90BB13` could be a valid regular rust int literal, or a `B13` literal. Maybe we should require a `_` before the suffix? We can also make it a requirement just for the (hex, bits) combination.

https://doc.rust-lang.org/reference/tokens.html#integer-literals

Reply:
> Maybe we should require a _ before the suffix?
yep, that sounds good.